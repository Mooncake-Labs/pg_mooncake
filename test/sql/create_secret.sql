SELECT mooncake.create_secret('mooncake-test-secret', 'S3', 'ASDfExaMPLEKeyID', 'ASDfExaMPLESECRET', '{"ENDPOINT":"s3.us-east-1.amazonaws.com", "REGION":"us-east-1"}');
TRUNCATE TABLE mooncake.secrets;

SELECT mooncake.create_secret('mooncake-test-secret', 'S3', 'ASDfExaMPLEKeyID', 'ASDfExaMPLESECRET', '{"ENDPOINT":"s3.us-east-1.amazonaws.com", "REGION":"us-east-1", "SCOPE":"s3://mooncakebucket/folder"}');
TRUNCATE TABLE mooncake.secrets;

SELECT mooncake.create_secret('mooncake-test-secret', 'S3', 'ASDfExaMPLEKeyID', 'ASDfExaMPLESECRET', '{"ENDPOINT":"s3.us-east-1.amazonaws.com", "REGION":"us-east-1", "USE_SSL":"FALSE"}');
TRUNCATE TABLE mooncake.secrets;

SELECT mooncake.create_secret('mooncake-test-secret', 'S3', 'ASDfExaMPLEKeyID', 'ASDfExaMPLESECRET', '{"ENDPOINT":"s3.us-east-1.amazonaws.com", "REGION":"us-east-1", "URL_STYLE":"path"}');
TRUNCATE TABLE mooncake.secrets;

SELECT mooncake.create_secret('mooncake-test-secret', 'S3', 'ASDfExaMPLEKeyID', 'ASDfExaMPLESECRET', '{"ENDPOINT":"s3express-use1-az6.us-east-1.amazonaws.com", "REGION":"us-east-1"}');
TRUNCATE TABLE mooncake.secrets;

SELECT mooncake.create_secret('mooncake-test-secret', 'S3', 'ASDfExaMPLEKeyID', 'ASDfExaMPLESECRET', '{"ENDPOINT":"s3.us-east-1.amazonaws.com", "REGION":"us-east-1", "URL":"path"}');
TRUNCATE TABLE mooncake.secrets;

SELECT mooncake.create_secret('mooncake-test-secret', 'S3', 'ASDfExaMPLEKeyID', 'ASDfExaMPLESECRET', '{"ENDPOINT":"https://s3.us-east-1.amazonaws.com", "REGION":"us-east-1"}');
TRUNCATE TABLE mooncake.secrets;

SELECT mooncake.create_secret('mooncake-test-secret', 'S3', 'ASDfExaMPLEKeyID', 'ASDfExaMPLESECRET', '{"ENDPOINT":"s3.us-east-1.amazonaws.com", "REGION":"us-east-1", "URL_STYLE":"xpath"}');
TRUNCATE TABLE mooncake.secrets;

SELECT mooncake.create_secret('mooncake-test-secret', 'XYZ', 'ASDfExaMPLEKeyID', 'ASDfExaMPLESECRET', '{"ENDPOINT":"s3.us-east-1.amazonaws.com", "REGION":"us-east-1"}');
TRUNCATE TABLE mooncake.secrets;
