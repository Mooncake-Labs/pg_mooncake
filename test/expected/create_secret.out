SELECT mooncake.create_secret('mooncake-test-secret', 'S3', 'ASDfExaMPLEKeyID', 'ASDfExaMPLESECRET', '{"ENDPOINT":"s3.us-east-1.amazonaws.com", "REGION":"us-east-1"}');
 create_secret 
---------------
 
(1 row)

TRUNCATE TABLE mooncake.secrets;
SELECT mooncake.create_secret('mooncake-test-secret', 'S3', 'ASDfExaMPLEKeyID', 'ASDfExaMPLESECRET', '{"ENDPOINT":"s3.us-east-1.amazonaws.com", "REGION":"us-east-1", "SCOPE":"s3://mooncakebucket/folder"}');
 create_secret 
---------------
 
(1 row)

TRUNCATE TABLE mooncake.secrets;
SELECT mooncake.create_secret('mooncake-test-secret', 'S3', 'ASDfExaMPLEKeyID', 'ASDfExaMPLESECRET', '{"ENDPOINT":"s3.us-east-1.amazonaws.com", "REGION":"us-east-1", "USE_SSL":"FALSE"}');
 create_secret 
---------------
 
(1 row)

TRUNCATE TABLE mooncake.secrets;
SELECT mooncake.create_secret('mooncake-test-secret', 'S3', 'ASDfExaMPLEKeyID', 'ASDfExaMPLESECRET', '{"ENDPOINT":"s3.us-east-1.amazonaws.com", "REGION":"us-east-1", "URL_STYLE":"path"}');
 create_secret 
---------------
 
(1 row)

TRUNCATE TABLE mooncake.secrets;
SELECT mooncake.create_secret('mooncake-test-secret', 'S3', 'ASDfExaMPLEKeyID', 'ASDfExaMPLESECRET', '{"ENDPOINT":"s3express-use1-az6.us-east-1.amazonaws.com", "REGION":"us-east-1"}');
 create_secret 
---------------
 
(1 row)

TRUNCATE TABLE mooncake.secrets;
SELECT mooncake.create_secret('mooncake-test-secret', 'S3', 'ASDfExaMPLEKeyID', 'ASDfExaMPLESECRET', '{"ENDPOINT":"s3.us-east-1.amazonaws.com", "REGION":"us-east-1", "URL":"path"}');
ERROR:  Invalid extra parameter: URL
Allowed parameters are ENDPOINT, REGION, SCOPE, URL_STYLE, USE_SSL.
TRUNCATE TABLE mooncake.secrets;
SELECT mooncake.create_secret('mooncake-test-secret', 'S3', 'ASDfExaMPLEKeyID', 'ASDfExaMPLESECRET', '{"ENDPOINT":"https://s3.us-east-1.amazonaws.com", "REGION":"us-east-1"}');
ERROR:  Invalid ENDPOINT format: https://s3.us-east-1.amazonaws.com
Use domain name excluding http prefix
TRUNCATE TABLE mooncake.secrets;
SELECT mooncake.create_secret('mooncake-test-secret', 'S3', 'ASDfExaMPLEKeyID', 'ASDfExaMPLESECRET', '{"ENDPOINT":"s3.us-east-1.amazonaws.com", "REGION":"us-east-1", "URL_STYLE":"xpath"}');
ERROR:  Invalid URL_STYLE: xpath
Allowed values for URL_STYLE are "path" or "vhost".
TRUNCATE TABLE mooncake.secrets;
SELECT mooncake.create_secret('mooncake-test-secret', 'XYZ', 'ASDfExaMPLEKeyID', 'ASDfExaMPLESECRET', '{"ENDPOINT":"s3.us-east-1.amazonaws.com", "REGION":"us-east-1"}');
ERROR:  Unsupported secret type: XYZ
Only secrets of type S3 are supported.
TRUNCATE TABLE mooncake.secrets;
